src/vs/code/browser/workbench/workbench.ts

```diff
@@ -32,15 +40,12 @@ class SecretStorageProvider implements ISecretStorageProvider {
 			extensionKey?.extensionId === 'membrane.membrane' &&
 			extensionKey?.key === 'membraneApiToken'
 		) {
-			// HACK: Find the first key that matches the pattern of auth0 React
-			const localStorageKey = Object.keys(localStorage).find((key) =>
-				key.includes('::default::openid')
-			);
-			if (localStorageKey) {
-				const json = localStorage.getItem(localStorageKey);
-				const value = JSON.parse(json!);
-				return value.body.access_token;
-			} else {
+			try {
+				if (!this.membraneToken) {
+					this.membraneToken = await this.getAuthToken();
+				}
+				return this.membraneToken;
+			} catch (error) {
```

Comment:
I don't think we should cache the token because auth0 might decide to refresh it in the middle of a session.

Reply:
Also, `error` is being ignored here. We're not gonna know what happened.